src/interp/interp.h

```diff
@@ -119,13 +119,44 @@ struct Table {
   std::vector<Index> func_indexes;
 };
 
+struct MemoryData {
+  size_t capacity_;
+  size_t size_;
+  char* data_;
+
+  MemoryData(size_t size_);
+  MemoryData(const MemoryData&) = delete;
+  MemoryData(MemoryData&&);
+
+  ~MemoryData();
+
+  MemoryData& operator=(const MemoryData&) = delete;
+  MemoryData& operator=(MemoryData&&);
+
+  char& operator[](size_t i) { return *(data_ + i); }
+  const char& operator[](size_t i) const { return *(data_ + i); }
+
+  size_t size() const noexcept { return size_; }
+
+  char* begin() noexcept { return data_; }
+  const char* cbegin() const noexcept { return data_; }
+
+  char* end() noexcept { return data_ + size_; }
+  const char* cend() const noexcept { return data_ + size_; }
+
+  char* data() noexcept { return data_; }
+  const char* data() const noexcept { return data_; }
+
+  void resize(size_t size);
+};
```

Comment:
Could we add a static assert to verify that `MemoryData` is truely standard layout?

Reply:
Fixed during the latest rebase